src/main/java/org/usfirst/frc4048/common/loggingv2/CommandLogger.java

```diff
@@ -0,0 +1,53 @@
+package org.usfirst.frc4048.common.loggingv2;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.*;
+
+public class CommandLogger {
+    private final LinkedHashMap<CommandKey, Queue<Boolean>> toLogCommandStatus = new LinkedHashMap<>();
+    private static final CommandLogger inst = new CommandLogger();
+    private boolean hasInit;
+
+    public static CommandLogger get() {
+        return inst;
+    }
+
+    private CommandLogger() {}
+
+    public void init() {
+        if (hasInit) return;
+        CommandScheduler.getInstance().onCommandInitialize(command -> {
+            Queue<Boolean> toLogBools = toLogCommandStatus.getOrDefault(command, new LinkedList<>());
+            toLogBools.add(true);
```

Comment:
Can’t this be shorter with computeifAbsent?

Reply:
Yes!